File: VB/S131978/Form1.vb
Imports DevExpress.XtraLayout
Imports System
Imports System.Data
Imports System.Windows.Forms

Namespace S131978
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			productsBindingSource.DataSource = GetProductsDataTable()
			DirectCast(layoutView1, ILayoutControl).OptionsFocus.MoveFocusDirection = MoveFocusDirection.DownThenAcross
		End Sub

		Private Function GetProductsDataTable() As DataTable
			Dim table As New DataTable()
			table.TableName = "Products"
			table.Columns.Add(New DataColumn("ProductID", GetType(Integer)))
			table.Columns.Add(New DataColumn("ProductName", GetType(String)))
			table.Columns.Add(New DataColumn("SupplierID", GetType(Integer)))
			table.Columns.Add(New DataColumn("CategoryID", GetType(String)))
			table.Columns.Add(New DataColumn("QuantityPerUnit", GetType(Integer)))
			table.Columns.Add(New DataColumn("UnitPrice", GetType(Double)))
			table.Columns.Add(New DataColumn("UnitsInStock", GetType(Integer)))
			table.Columns.Add(New DataColumn("UnitsOnOrder", GetType(Integer)))
			table.Columns.Add(New DataColumn("ReorderLevel", GetType(Integer)))
			For i As Integer = 0 To 19
				Dim index As Integer = i + 1
				table.Rows.Add(index, "Product " & index, index * 2, index * 3, index Mod 7, index * 1000R + index * 100R, index Mod 5, index Mod 9, index Mod 3)
			Next i
			Return table
		End Function
	End Class
End Namespace

File: CS/S131978/Form1.cs
using DevExpress.XtraLayout;
using System;
using System.Data;
using System.Windows.Forms;

namespace S131978
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            productsBindingSource.DataSource = GetProductsDataTable();
            ((ILayoutControl)layoutView1).OptionsFocus.MoveFocusDirection = MoveFocusDirection.DownThenAcross;
        }

        DataTable GetProductsDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "Products";
            table.Columns.Add(new DataColumn("ProductID", typeof(int)));
            table.Columns.Add(new DataColumn("ProductName", typeof(string)));
            table.Columns.Add(new DataColumn("SupplierID", typeof(int)));
            table.Columns.Add(new DataColumn("CategoryID", typeof(string)));
            table.Columns.Add(new DataColumn("QuantityPerUnit", typeof(int)));
            table.Columns.Add(new DataColumn("UnitPrice", typeof(double)));
            table.Columns.Add(new DataColumn("UnitsInStock", typeof(int)));
            table.Columns.Add(new DataColumn("UnitsOnOrder", typeof(int)));
            table.Columns.Add(new DataColumn("ReorderLevel", typeof(int)));
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                table.Rows.Add(index, "Product " + index, index * 2, index * 3, index % 7, index * 1000d + index * 100d, index % 5, index % 9, index % 3);
            }
            return table;
        }
    }
}